File: Editorial/2078_Two_Furthest_Houses_With_Different_Colors/C/2_Greedy.c
int maxDistance(int* colors, int colorsSize) {
    int n = colorsSize;
    int ans = 0;
    for (int i = 0; i < n - 1; i++) {
        if (colors[i] != colors[n - 1]) {
            int d = i > n - 1 - i ? i : n - 1 - i;
            if (d > ans) ans = d;
        }
    }
    return ans;
}

File: Editorial/2078_Two_Furthest_Houses_With_Different_Colors/Go/2_Greedy.go
func maxDistance(colors []int) int {
	n := len(colors)
	ans := 0
	for i := 0; i < n-1; i++ {
		if colors[i] != colors[n-1] {
			d := i
			if n-1-i > d {
				d = n - 1 - i
			}
			if d > ans {
				ans = d
			}
		}
	}
	return ans
}
